### DISCUSSION TITLE
Is it necessary to create settings file in every repository?

### DISCUSSION THREAD

**Discussion by mitchboulay** [repository-settings/app#542] [Q&A]
Hi,
I have my repository settings configured in org/.github/.github/settings.yml and I am trying to configure such that on repository creation, each repository will automatically inherit the configuration.

Is this possible, or is is expected that for each repo that I want to inherit settings, I must add a .github/settings.yml & _extends?

**Comment by gr2m** [comment]
No need to create a `.github/settings.yml` in each file. The app inherits the setting from the `[org]/.github` repository by default

**Reply by mitchboulay** [reply]
> No need to create a `.github/settings.yml` in each file. The app inherits the setting from the `[org]/.github` repository by default

Hmm I can't seem to get it to work. Does the repo have to be public?

**Reply by gr2m** [reply]
The app needs to be installed on both repositories

**Reply by mitchboulay** [reply]
Okay, that makes sense. I thought having the app installed in my organization would have been enough.

**Reply by gr2m** [reply]
Unfortunately no, the app needs to have file read access in order to read the configuration, at least on private repositories. I think public repositories should work to read configuration without the need to install the app

**Reply by mitchboulay** [reply]
But I have that under my app configuration? Regardless of whether my org/.github repo is public or private, the settings are only being invoked when that .github/settings.yml file extends .github.
![Screen Shot 2022-01-12 at 5 04 11 PM](https://user-images.githubusercontent.com/48683522/149230322-87438e8c-3816-4900-b121-e1a55a4161f8.png)

**Reply by gr2m** [reply]
If your repository has a `.github/[app].yml` file, then you need to set the `_extends` key. If the file is not present, it will be loaded from `[org]/.github` at `.github/[app].yml`

See https://github.com/probot/octokit-plugin-config/#the-_extends-key

If that still doesn't work for you I'll  try to replicate the problem with a new app

**Reply by mitchboulay** [reply]
Okay, you are correct, I was able to get that to work as I can see the homepage URL appears to be pulling in and team configuration is applying. Have you, last one, any particular reason why branch rules wouldn't be inherited?

Below are my rules, again if I use the _extends key it works:
![Screen Shot 2022-01-13 at 12 24 11 PM](https://user-images.githubusercontent.com/48683522/149378740-56068171-c1ce-490a-ad08-93bfdbca8469.png)

**Reply by mitchboulay** [reply]
Please disregard. I created another repository and added a readme.md file on creation. I've gathered the following for this to work as I was expecting:
1. Settings must be defined prior to repository creation
2. Select option to add README.md so that the main branch is published and settings can be applied.

Thank you for your help with this @gr2m !
